lib/hamster/hash.rb

```diff
@@ -590,6 +590,25 @@ def fetch_values(*wanted)
       Vector.new(array.freeze)
     end
 
+    # Return the value of successively indexing into a hash
+    # If any of the keys is not present in the hash, return `nil`
+    #
+    # @example
+    #   h = Hamster::Hash[:a => 9, :b => Hamster::Hash[:c => 'a', :d => 4], :e => nil]
+    #   h.dig(:b, :c) # => "a"
+    #   h.dig(:b, :f) # => nil
+    # @params keys to fetch from the hash
+    # @return [Object]
+    def dig(key, *rest)
+      value = self[key]
+      if rest.empty? || value.nil?
+        value
+      elsif value.respond_to?(:dig)
+        value.dig(*rest)
```

Comment:
It does at least on the ruby 2.3.0 I have:

```shell
$ ruby --version
ruby 2.3.0dev (2015-11-17 trunk 52623) [x86_64-darwin15]

adam at machine in ~ 
$ irb
irb(main):001:0> {a: 9}.dig(:a, :b)
=> nil
```

Interestingly, it actually works on any class that implements dig:
```ruby
irb(main):001:0> class Diggable
irb(main):002:1>   def dig(*args)
irb(main):003:2>     'foo'
irb(main):004:2>   end
irb(main):005:1> end
=> :dig
irb(main):006:0> d = Diggable.new
=> #<Diggable:0x007f818316a358>
irb(main):007:0> {a: d}.dig(:a, :b)
=> "foo"
```

However, you do have to dig based on a key that the object can respond to:
```ruby
irb(main):008:0> {a: [1,2,3]}.dig(:a, 1)
=> 2
irb(main):009:0> {a: [1,2,3]}.dig(:a, :b)
TypeError: no implicit conversion of Symbol into Integer
	from (irb):9:in `dig'
	from (irb):9
	from /Users/adam/.rbenv/versions/2.3.0-dev/bin/irb:11:in `<main>'
```